server/src/main/java/org/elasticsearch/common/util/BigIntArray.java

```diff
@@ -40,6 +41,26 @@ final class BigIntArray extends AbstractBigArray implements IntArray {
         }
     }
 
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (size > Integer.MAX_VALUE / Integer.BYTES) {
+            throw new IllegalArgumentException();
+        }
+        int intSize = (int) size;
+        out.writeVInt(intSize * 4);
+        for (int i = 0; i < pages.length - 1; i++) {
+            out.write(pages[i]);
+        }
+        int end = intSize % INT_PAGE_SIZE;
+        out.write(pages[pages.length - 1], 0, end * Integer.BYTES);
+    }
+
+    static {
```

Comment:
Nit: maybe put this at the beginning of the class file, it's weird to have the static initializer somewhere in the middle of the file isn't it? :)